processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -873,127 +941,67 @@ private static void makeDimColumn(
-                  .create(vals)
-                  .transform(
-                      new Function<List<Integer>, VSizeIndexedInts>()
-                      {
-                        @Override
-                        public VSizeIndexedInts apply(List<Integer> input)
-                        {
-                          if (input == null) {
-                            return VSizeIndexedInts.fromList(
-                                new AbstractList<Integer>()
-                                {
-                                  @Override
-                                  public Integer get(int index)
-                                  {
-                                    return 0;
-                                  }
-
-                                  @Override
-                                  public int size()
-                                  {
-                                    return 1;
-                                  }
-                                }, dictionarySize
-                            );
-                          }
-                          return VSizeIndexedInts.fromList(
-                              input,
-                              dictionarySize
-                          );
-                        }
+              Iterables.transform(
+                  vals,
+                  new Function<List<Integer>, VSizeIndexedInts>()
+                  {
+                    @Override
+                    public VSizeIndexedInts apply(List<Integer> input)
+                    {
+                      if (input == null) {
+                        return VSizeIndexedInts.fromList(ImmutableList.<Integer>of(0), dictionarySize);
+                      } else {
+                        return VSizeIndexedInts.fromList(input, dictionarySize);
                       }
-                  )
+                    }
+                  }
+              )
           );
         }
       } else {
         final int dictionarySize = dictionary.size();
         singleValCol = null;
         multiValCol = VSizeIndexed.fromIterable(
-            FunctionalIterable
-                .create(vals)
-                .transform(
-                    new Function<List<Integer>, VSizeIndexedInts>()
-                    {
-                      @Override
-                      public VSizeIndexedInts apply(List<Integer> input)
-                      {
-                        return VSizeIndexedInts.fromList(
-                            input,
-                            dictionarySize
-                        );
-                      }
-                    }
-                )
+            Iterables.transform(
+                vals,
+                new Function<List<Integer>, VSizeIndexedInts>()
+                {
+                  @Override
+                  public VSizeIndexedInts apply(List<Integer> input)
+                  {
+                    return VSizeIndexedInts.fromList(input, dictionarySize);
```

Comment:
out of curiosity, why the behavior will change. "input" here is going to have only non-null elements, so NullsAtZeroConvertingIntList(input, true) will just be a noop. And, I agree now, it is ok to keep in the current form as NullsAtZeroConvertingIntList(input, true) not doing anything.